### PR TITLE
Generate metadata

### PR DESCRIPTION
**Pull request by wer1st**
Created empty lib that triggers the generation of metadata every build.

Signed-off-by: Mikhail Tagirov <dev.mikhail.tagirov@gmail.com>

### CODE REVIEW

File: parachain-gen/build.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .to_str()
+        .unwrap()
+        .to_string();
+    println!("cargo:rerun-if-changed={}/runtime", workspace_root);
+    println!("cargo:rerun-if-changed={}/pallets", workspace_root);
+    println!("cargo:rerun-if-changed={}/common", workspace_root);
```
Comment (vovac12) [inline parachain-gen/build.rs]: We don't have common in parachain repo

**Review by vovac12 — CHANGES_REQUESTED** [review]
I think we can add parachain_metadata.scale to .gitignore and add it as artifact to CI

File: parachain-gen/build.rs (hunk 1 above)
Comment (wer1st) [reply parachain-gen/build.rs]: But we will after #57 PR

File: parachain-gen/build.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+fn main() {
+    let metadata: Bytes = OpaqueMetadata::new(parachain_template_runtime::Runtime::metadata().into()).into();
+    let out_dir = PathBuf::from_str(&std::env::var("CARGO_MANIFEST_DIR").unwrap())
+        .unwrap()
+        .join("src/bytes");
+    std::fs::create_dir_all(&out_dir).unwrap();
+    std::fs::write(out_dir.join("parachain_metadata.scale"), metadata.0).unwrap();
+    let workspace_root = PathBuf::from_str(&std::env::var("CARGO_MANIFEST_DIR").unwrap())
+        .unwrap()
```
Comment (N1ghtStorm) [inline parachain-gen/build.rs]: Are all these unwraps required? Maybe it is better to use expect?

Comment (wer1st) [reply parachain-gen/build.rs]: Trying to be consistent with other our metadata generators)

**Review by N1ghtStorm — APPROVED** [review]

File: parachain-gen/build.rs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+		.parent()
+		.unwrap()
+		.parent()
+		.unwrap()
+		.to_str()
+		.unwrap()
+		.to_string();
+	println!("cargo:rerun-if-changed={}/runtime", workspace_root);
+	println!("cargo:rerun-if-changed={}/pallets", workspace_root);
+	println!("cargo:rerun-if-changed={}/common", workspace_root);
```
Comment (vovac12) [inline parachain-gen/build.rs:23]: We don't have this folder in this repo

**Review by vovac12 — CHANGES_REQUESTED** [review]

File: parachain-gen/build.rs (hunk 3 above)
Comment (wer1st) [reply parachain-gen/build.rs:23]: But we will have after #57 PR

**Review by vovac12 — APPROVED** [review]

**Pull request merged**